core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -873,12 +891,18 @@ public Builder setCurrentSchema(int schemaId) {
 
       this.currentSchemaId = schemaId;
 
-      changes.add(new MetadataUpdate.SetCurrentSchema(schemaId));
+      if (lastAddedSchemaId != null && lastAddedSchemaId == schemaId) {
+        // use -1 to signal that current was set to the last added schema
+        changes.add(new MetadataUpdate.SetCurrentSchema(-1));
```

Comment:
Would this be something to make as a constant? Perhaps `USE_LAST_ADDED_SCHEMA = -1`?

EDIT: Looking at further code, we use the same constant for partition spec and many other things. So maybe this comment would be better placed above the builder class or something? Or using a more generic constant like `private static final int KEEP_EXISTING_ID = -1`?

I'm overall ok with this though and like the idea of using `-1` this way.

Reply:
Added `LAST_ADDED`. Good suggestion, it's easier to read.